spec/models/reports/prescription_spec.rb

```diff
@@ -0,0 +1,157 @@
+        DrugLookup::CleanMedicineToDosage.create({medicine: "Telmisartan", dosage: 80, rxcui: 444})
+
+        DrugLookup::RawToCleanMedicine.create({raw_name: "Aten", raw_dosage: "10mg", rxcui: 555})
+        DrugLookup::CleanMedicineToDosage.create({medicine: "Atenolol", dosage: 10, rxcui: 555})
+
+        patient = FactoryBot.create(:patient, recorded_at: june_2021[:two_years_ago])
+        add_drug(patient, "Amlo", "5mg", june_2021[:ten_months_ago])
+        add_drug(patient, "Amlo", "5mg", june_2021[:nine_months_ago])
+        add_drug(patient, "Amlo", "10mg", june_2021[:eight_months_ago])
+        add_drug(patient, "Amlo", "10mg", june_2021[:seven_months_ago])
+        # does not visit for 2 months
+        add_drug(patient, "Telmi", "40mg", june_2021[:four_months_ago])
+
+        add_drug(patient, "Amlo", "10mg", june_2021[:three_months_ago])
+        add_drug(patient, "Telmi", "40mg", june_2021[:three_months_ago])
+        # does not visit for 1 month
+        add_drug(patient, "Telmi", "80mg", june_2021[:one_month_ago])
+        add_drug(patient, "Aten", "10mg", june_2021[:one_month_ago])
+
+        described_class.refresh
+
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:eleven_months_ago]).amlodipine).to eq(0)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:eleven_months_ago]).telmisartan).to eq(0)
+
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:ten_months_ago]).amlodipine).to eq(5)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:nine_months_ago]).amlodipine).to eq(5)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:eight_months_ago]).amlodipine).to eq(10)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:seven_months_ago]).amlodipine).to eq(10)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:six_months_ago]).amlodipine).to eq(10)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:five_months_ago]).amlodipine).to eq(10)
+
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:four_months_ago]).amlodipine).to eq(10)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:four_months_ago]).telmisartan).to eq(40)
+
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:three_months_ago]).amlodipine).to eq(10)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:three_months_ago]).telmisartan).to eq(40)
+
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:two_months_ago]).amlodipine).to eq(10)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:two_months_ago]).telmisartan).to eq(40)
+
+        # expect(described_class.find_by(patient: patient, month_date: june_2021[:one_month_ago]).other_bp_medications).to eq(10)
```

Comment:
👾

Reply:
Fixed in ea8901efcc791aea90a061273751826412afb679.